modules/picard_qc/1.0/picard_qc.smk

```diff
@@ -0,0 +1,323 @@
+        {params.opts} 
+        I={input.bam} O={output.metrics} H={output.histogram}
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+# capture metrics
+rule _picard_qc_hs_metrics:
+    input:
+        bam = rules._picard_qc_input_bam.output.bam,
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa"),
+        intervals = CFG["inputs"]["intervals"]
+    output:
+        hs = CFG["dirs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/hs_metrics",
+        intervals = CFG["dirs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/interval_hs_metrics"
+    log:
+        stdout = CFG["logs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/hs_metrics.stdout.log",
+        stderr = CFG["logs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/hs_metrics.stderr.log"
+    params:
+        opts = CFG["options"]["hs_metrics"]
+    conda:
+        CFG["conda_envs"]["picard"]
+    threads:
+        CFG["threads"]["hs_metrics"]
+    resources:
+        mem_mb = CFG["mem_mb"]["hs_metrics"]
+    shell:
+        op.as_one_line("""
+        picard -Xmx{resources.mem_mb}m CollectHsMetrics {params.opts} 
+        I={input.bam} O={output.hs} PER_TARGET_COVERAGE={output.intervals} 
+        R={input.fasta} TI={input.intervals} BI={input.intervals} 
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+# mRNA metrics
+rule _picard_qc_rrna_int:
+    input:
+        bam = rules._picard_qc_input_bam.output.bam,
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa"),
+        rrna_int = reference_files(CFG["inputs"]["rrna_intervals"])
+    output:
+        rrna_int = CFG["dirs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/rrna_int_list"
+    log:
+        stderr = CFG["logs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/rrna_int.stderr.log"
+    conda:
+        CFG["conda_envs"]["samtools"]
+    shell:
+        op.as_one_line("""
+        samtools view -H {input.bam} | 
+        grep -v "@RG" | cat - {input.rrna_int} 
+        > {output.rrna_int}
+        2> {log.stderr}
+        """)
+
+
+rule _picard_qc_rnaseq_metrics:
+    input:
+        bam = rules._picard_qc_input_bam.output.bam,
+        rrna_int = rules._picard_qc_rrna_int.output.rrna_int,
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa"),
+        refFlat = reference_files(CFG["inputs"]["refFlat"])
+    output:
+        metrics = CFG["dirs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/rnaseq_metrics"
+    log:
+        stdout = CFG["logs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/rnaseq_metrics.stdout.log",
+        stderr = CFG["logs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/rnaseq_metrics.stderr.log"
+    params:
+        opts = CFG["options"]["rnaseq_metrics"]["base"],
+        strand = op.switch_on_column("strand", CFG["samples"], CFG["options"]["rnaseq_metrics"]["strand"], match_on = "sample")
```

Comment:
First, good discovery of the `match_on` parameter!

Second, you're going to want to create a new schema file in the root `schemas/` directory and symlink it into the module's `schemas/` directory. You should probably make it an enum with only three (?) options: forward, reverse, or unstranded. Whatever terms you choose, update your switch accordingly.

Reply:
I realise if I do that then I will have to provide a non-empty `strand` column even if I'm not preocessing mrna samples. Would it be reasonable to convert all the `Nan` to `"-"`, which would not be evaluated for non-`mrna` samples and evaluated as unstranded` for `mrna` samples`